### PR TITLE
discovery: Normalize service name

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Normalize the service name using the helper functions from pkg/trace.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1221

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/service_detector_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/service_detector_test.go:4]:
```suggestion
// Copyright 2024-present Datadog, Inc.
```

File: pkg/collector/corechecks/servicediscovery/service_detector_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/language"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/usm"
+)
+
+func TestFixup(t *testing.T) {
+	meta := fixupMetadata(usm.ServiceMetadata{Name: "Foo", DDService: "Bar"}, language.Go)
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/service_detector_test.go:18]: nit - to cover more cases (and not handling only first letter)
```suggestion
	meta := fixupMetadata(usm.ServiceMetadata{Name: "fOo", DDService: "BAR"}, language.Go)
```

**Review by guyarb — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/service_detector_test.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/service_detector_test.go:4]: discovery: Fix copyright year

**Pull request merged**